spec/acceptance/init_spec.rb

```diff
@@ -20,8 +20,27 @@ class { 'bolt': use_release_package => false }
     end
   end
 
-  describe command('yum clean all --verbose; rm -rf /etc/yum.repos.d/puppet-tools-release.repo; yum erase --assumeyes puppet-bolt puppet-tools-release') do
-    its(:exit_status) { is_expected.to eq 0 }
+  describe 'with use_release_package=false & ensure=absent' do
+    it_behaves_like 'an idempotent resource' do
+      let(:manifest) do
+        <<-PUPPET
+        class { 'bolt':
+          use_release_package => false,
+          version => 'absent',
+        }
+        PUPPET
+      end
+    end
+
+    describe 'packages' do
+      it { expect(package('puppet-bolt')).not_to be_installed }
+      it { expect(package('puppet-tools-release')).not_to be_installed }
+    end
+
+    # puppet 7 and below remove repos from the file, puppet 8 deletes the file
+    describe file('/etc/yum.repos.d/puppet-tools.repo') do
+      it { is_expected.to be_missing.or(have_attributes(size: 0)) }
+    end
```

Comment:
Ideally this would also be updated to the modern syntax, but not blocking:
```suggestion
    it { expect(file('/etc/yum.repos.d/puppet-tools.repo')).to be_missing.or(have_attributes(size: 0)) }
```